services/visual_recognition/v3.js

```diff
@@ -0,0 +1,348 @@
+
+        var visualRecognition = watson.visual_recognition({
+          api_key: apikey,
+          version: 'v3',
+          version_date: '2016-05-19'
+        });
+
+    if (feature == 'classifyImage')
+    {
+      visualRecognition.classify(params, cbdone);
+    } else if (feature === 'detectFaces') {
+      visualRecognition.detectFaces(params, cbdone);
+    } else if (feature === 'recognizeText') {
+      visualRecognition.recognizeText(params, cbdone);
+    } else if (feature === 'createClassifier') {
+      visualRecognition.createClassifier(params, cbdone);
+    } else if (feature === 'retrieveClassifiersList') {
+      visualRecognition.listClassifiers(params, cbdone);
+    } else if (feature === 'retrieveClassifierDetails') {
+      visualRecognition.getClassifier(params, cbdone);
+    } else if (feature === 'deleteClassifier') {
+      visualRecognition.deleteClassifier(params, cbdone);
+    }
+
+  }
+
+  // This is the Watson Visual Recognition V3 Node
+  function WatsonVisualRecognitionV3Node (config) {
+    RED.nodes.createNode(this, config);
+    var node = this;
+    var message;
+    this.on('input', function (msg) 
+    {
+      // Check which single feature has been requested.
+      var feature = config["image-feature"];
+      node.status({});
+      temp.cleanup(); // so there is at most 1 temp file at a time (did not found a better solution...)
+
+      if (!msg.payload) {
+        this.status({fill:'red', shape:'ring', text:'missing payload'}); 
+        message = 'Missing property: msg.payload';
+        node.error(message, msg);
+        return;
+      }
+      if (feature != 'retrieveClassifiersList' && feature != 'retrieveClassifierDetails' && feature != 'deleteClassifier')
+      if (typeof msg.payload == "boolean" || typeof msg.payload == "number") {
+          this.status({fill:'red', shape:'ring', text:'bad format payload'}); 
+          message = 'Bad format : msg.payload must be a URL string or a Node.js Buffer';
+          node.error(message, msg);
+          return;
+        }
+
+      // If it is present the newly provided user entered key takes precedence over the existing one. 
+      apikey = s_apikey || this.credentials.apikey;
+      this.status({}); 
+
+      if (!apikey) {
+        this.status({fill:'red', shape:'ring', text:'missing credentials'});          
+        message ='Missing Watson Visual Recognition API service credentials'; 
+        node.error(message, msg);
+        return;
+      }
+
+      var params = {};
+
+      // This is the callback after the call to the watson service.    
+      // Set up as a var within this scope, so it has access to node, msg etc. 
+      // in preparation for the Watson service action       
+      var actionComplete = function(err, body, other) {
+
+        if (err != null && body==null)
+        {
+          node.status({fill:'red', shape:'ring', text:'call to watson visual recognition v3 service failed'}); 
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Line is too long.](https://www.codacy.com/app/BetaWorks-NodeRED-Watson/node-red-node-watson/file/2686279757/issues/source?bid=3330694&fileBranchId=3336646#l162)